src/Compilers/CSharp/Portable/Lowering/IteratorRewriter/IteratorRewriter.cs

```diff
@@ -115,31 +115,31 @@ protected bool VerifyPresenceOfRequiredAPIs()
             }
 
             bool hasErrors = bag.HasAnyErrors();
-            if (hasErrors)
+            if (!hasErrors)
             {
-                diagnostics.AddRange(bag);
+                bag.DiagnosticBag.Clear();
```

Comment:
Same clarifying comment would be useful.

Reply:
> Same clarifying comment would be useful.

Moving to using AddDependencies API instead 

---
In reply to: [362137708](https://github.com/dotnet/roslyn/pull/40560#discussion_r362137708) [](ancestors = 362137708)

Reply:
This is addressed in [https://github.com/dotnet/roslyn/pull/40640](https://github.com/dotnet/roslyn/pull/40640)

---
In reply to: [362138975](https://github.com/dotnet/roslyn/pull/40560#discussion_r362138975) [](ancestors = 362138975,362137708)